[DM-4788]: Added user docs and fragment lib update for the new provided option for files in software and firmware repository items. (#3472)
SoftwareAG/c8y-docs: content/device-integration/fragment-library-bundle/firmware.md

`115,ubuntu core,20.04.3,http://test.com`

> ### Using $PROVIDED firmware URLs {#provided-firmware}

Reviewer: This section should be moved to below the explanation of the regular FW updates as an additional explanation

> ### Using `$PROVIDED` firmware URLs {#provided-firmware}

Reviewer 2: ```suggestion
### Using $PROVIDED firmware URLs {#provided-firmware}
```

If the `url` is set to `$PROVIDED`, no firmware file is uploaded or linked from the platform. In this case, the device is fully responsible for resolving the correct binary. If the device cannot resolve the binary, the operation must be marked as **FAILED**.

This option is useful when the device can retrieve the correct firmware file dynamically based on its environment, architecture, or internal logic.

**SmartREST example**

> To mark a firmware as provided:

Reviewer: There is nothing to mark the firmware here, this is an operation that is being sent to the device. I would remove the SR example section here. Readers can read the other examples and then infer that $PROVIDED is sometimes used instead of regular URLs.

  `115,ubuntu core,20.04.3,http://test.com`
6. Set operation status to SUCCESSFUL<br>
  `503,c8y_Firmware`

{{< c8y-admon-info >}}

> If the URL is set to `$PROVIDED`, no firmware file is uploaded or linked from the platform. In this case, the device is fully responsible for resolving the correct binary. If the device cannot resolve the binary, the operation is marked as **FAILED**.

Reviewer: ```suggestion
If the URL is set to `$PROVIDED`, no firmware file is uploaded or linked from the platform. In this case, the device is fully responsible for resolving the correct binary. If the device cannot resolve the binary, it must mark the operation as **FAILED**.
```

> If the `url` is set to `$PROVIDED`, no firmware file is uploaded or linked from the platform. In this case, the device is fully responsible for resolving the correct binary. If the device cannot resolve the binary, the operation must be marked as **FAILED**.

Reviewer 2: ```suggestion
If the URL is set to `$PROVIDED`, no firmware file is uploaded or linked from the platform. In this case, the device is fully responsible for resolving the correct binary. If the device cannot resolve the binary, the operation is marked as **FAILED**.
```

> This option is useful when the device can retrieve the correct firmware file dynamically based on its environment, architecture, or internal logic.

Reviewer 2: ```suggestion
This option is useful if the device can retrieve the correct firmware file dynamically based on its environment, architecture, or internal logic.
```
